### DISCUSSION TITLE
New pipeline events

### DISCUSSION THREAD

**Discussion by anbraten** [woodpecker-ci/woodpecker#3425] [Request a Feature]
We may add new pipeline events in the future:

Possible events:
- [x] PR close #286
- [x] Release (type: draft, pre, stable; action: created, updated, deleted) #764 
- [x] Cron #8 #934 
- [x] Support manually triggering a pipeline #83 #240 
- [ ] comments (create, delete, update) / #814 
- [ ] issues (created, updated, closed, reopened)
- [ ] pull request updated (title, description, milestone, labels added removed, assignee) ( #5214 or partially #5155 )

**Comment by anbraten** [comment]
Proposal: Adding comments event

Actions:
- `comment`: created & updated
- `comment_deleted` deleted

Content:
- author
- body
- date
- context (type: issue / pr, url)
- reactions?

**Comment by anbraten** [comment]
Proposal: Adding issue event

Events:
- `issue`: created & updated
- `issue_closed`: closed & deleted

Contents:
- title
- description
- labels
- assignee
- milestone
- reactions

**Comment by anbraten** [comment]
Proposal: Adding pull-request event

Events:
- `pull_request`: created, pushed new commits
- `pull_request_updated`: (changed title, description, ...) #5155 
- `pull_request_closed`

Contents:
- commits
- source / target branch
- changed_files
- title
- description
- labels
- assignee
- milestone
- reactions

**Reply by pat-s** [reply]
https://codeberg.org/crowci/crow/pulls/82 -> `pull_request_edited`

**Reply by 6543** [reply]
started at https://github.com/woodpecker-ci/woodpecker/pull/5214 and would need input if this will be accepted ... if so I will finish it :)

**Comment by pixel-fabian** [comment]
I'd really love to see `pushed new commits (event: pull_request)`. I like the workflow to execute a pipeline in feature branches on pull_reqeust, but if a later commit will fix a failing pipeline i currently see no elegant method to execute it again in the PR.

**Reply by qwerty287** [reply]
That is supported already if I understand you point. If you push new commits to a pull request, that will trigger a `pull_request` event.

**Reply by pixel-fabian** [reply]
That is the behaviour i was expecting. Yesterday it did not pick up a new commit in the PR but today it worked fine.

**Reply by erdnaxeli** [reply]
It seems to not work with push force, at least with gitea.

edit: sometimes it does not work, sometimes it does :grimacing:
